src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+
+import './TaskTitle.css';
+
+class TaskTitle extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newTitle: '',
+      showInput: false
+    };
+  }
+
+  componentWillReceiveProps = (newProps) => {
+    // handle case when selected task will change.
+    if (newProps.title !== this.props.title) {
+      this.setState({showInput: false});
+    }
+  }
+
+  validTitle = (value) => {
+    return value !== ''
+  }
+
+  toggleInput = () => {
+    const { showInput } = this.state;
+    this.setState({showInput: !showInput});
+  }
+
+  onEnter = (value) => {
+    const { id, patchTask } = this.props;
+    // const { newTitle } = this.state;
+    patchTask({
+      gid: id,
+      title: value
+    })
+  }
+
+  onChange = (e) => {
+    const { value } = e.target;
+    const { key } = e;
```

Comment:
@colinfruit Could do `const { key, target: { value } = e;`, FYI :-) but this is fine :+1: